provider/datagen/src/lib.rs

```diff
@@ -315,6 +319,9 @@ impl Default for BakedOptions {
 #[non_exhaustive]
 pub enum Out {
     /// Output to a file system tree
+    ///
+    /// Only available with the `icu_provider_fs` feature
```

Comment:
Nit:

```suggestion
    /// **Enabled with the `icu_provider_fs` Cargo feature.**
```

As discussed previously, a large percentage of problems our users have is when they don't have the right features enabled, so I think we should try to continue to be loud and consistent on functions that need a Cargo feature.